### PR TITLE
* Session changes

### PR DESCRIPTION
**Pull request by saynb**
* Adding a create function that needs to be overridden
  * Deleting default ctor for Session
  * Changing handleGet so that a temporary can be returned
* TableData changes
  * Adding reset APIs and implementation
  * Adding removeActiveFields so that active fields could be removed

### CODE REVIEW

File: src/tdi_table_data.cpp [hunk 1]
```diff
@@ -161,11 +184,21 @@ tdi_status_t TableData::getParent(const tdi::Learn ** /*learn*/) const {
...
                                  bool *is_active) const {
-  *is_active =
-      std::find(active_fields_.begin(), active_fields_.end(), field_id) !=
-      active_fields_.end();
+  auto it = this->removed_one_ofs_.find(field_id);
+  if (it != this->removed_one_ofs_.end()) {
+    *is_active = false;
+    return TDI_SUCCESS;
+  }
+
```
Comment (saynb) [inline src/tdi_table_data.cpp:195]: Check `all_fields_set_` and return true

File: include/tdi/common/tdi_table_data.hpp [hunk 2]
```diff
@@ -359,6 +360,33 @@ class TableData {
...
+  /**
+   * @brief Data object reset
+   *
+   * @param[in] action_id Action ID. Can be put in as 0 if actions aren't
+   * supported by the table or if it is unknown.
+   * @param[in] fields Vector of field IDs
+   *
+   * @return Status of the API call
+   */
+  tdi_status_t reset(const tdi_id_t &action_id,
```
Comment (saynb) [inline include/tdi/common/tdi_table_data.hpp:372]: make virtual

File: include/tdi/common/tdi_session.hpp [hunk 3]
```diff
@@ -165,10 +166,12 @@ class Session {
   virtual tdi_status_t abortTransaction() const = 0;
   /** @} */  // End of group Transaction
  protected:
+  virtual tdi_status_t create() = 0;
```
Comment (saynb) [inline include/tdi/common/tdi_session.hpp:169]: make public

**Review by swaroopsarma — APPROVED** [review]
looks good to me

File: src/tdi_table_data.cpp [hunk 4]
```diff
@@ -161,11 +184,25 @@ tdi_status_t TableData::getParent(const tdi::Learn ** /*learn*/) const {
   LOG_ERROR("%s:%d Not supported", __func__, __LINE__);
   return TDI_NOT_SUPPORTED;
 }
+
 tdi_status_t TableData::isActive(const tdi_id_t &field_id,
                                  bool *is_active) const {
-  *is_active =
-      std::find(active_fields_.begin(), active_fields_.end(), field_id) !=
-      active_fields_.end();
+  auto it = this->removed_one_ofs_.find(field_id);
```
Comment (weiqiand) [inline src/tdi_table_data.cpp:190]: can you add comments for removed_one_ofs_ in terms all_fields_set_ what happens for all_fields_set_ is True? Should we check the all_field_set_ first or not?

**Review by weiqiand — APPROVED** [review]

**Pull request merged**